test/com/yetu/oauth2provider/services/data/UserServiceSpec.scala

```diff
@@ -1,75 +1,72 @@
 package com.yetu.oauth2provider.services.data
 
 import com.yetu.oauth2provider.base.DataServiceBaseSpec
-import com.yetu.oauth2provider.registry.{ TestRegistry, IntegrationTestRegistry }
-import com.yetu.oauth2provider.services.data.interface.IPersonService
+import com.yetu.oauth2provider.registry.{ IntegrationTestRegistry, TestRegistry }
 import com.yetu.oauth2provider.utils.DateUtility._
 import org.joda.time.DateTime
+import org.scalatest.concurrent.ScalaFutures
+import org.scalatest.time.{ Seconds, Span }
 import securesocial.core.PasswordInfo
 import securesocial.core.services.SaveMode
 
+import scala.concurrent.ExecutionContext.Implicits.global
+
 /*
  * This test class is extended below to run the same tests against the in-memory and the LDAP implementations.
  */
-abstract class UserServiceBase extends DataServiceBaseSpec {
-
-  override def beforeEach {
-    personService.deleteUser(testUser.identityId.userId)
-  }
-
-  override def afterEach {
-    personService.deleteUser(testUser.identityId.userId)
-  }
+abstract class UserServiceBase extends DataServiceBaseSpec with ScalaFutures {
 
   s"The [$databaseImplementationName] User Service" must {
 
-    "store and retrieve a user " in {
-      personService.save(testUser.toBasicProfile, SaveMode.SignUp)
-      val Some(yetuUser) = personService.findYetuUser(testUser.userId)
-
-      yetuUser.email mustEqual testUser.email
-      yetuUser.firstName mustEqual testUser.firstName
-      yetuUser.passwordInfo mustEqual testUser.passwordInfo
-      dateToString(yetuUser.userAgreement.get.acceptTermsAndConditionsDate.toDate) mustEqual dateToString(DateTime.now().toDate)
-      yetuUser.userAgreement.get.acceptTermsAndConditions mustEqual true
+    "store and retrieve a user" in {
+
+      val result = for {
+        delete <- personService.deleteUser(testUser.userId)
+        save <- personService.save(testUser, SaveMode.SignUp)
+        retrieve <- personService.findUser(testUser.userId)
+      } yield retrieve
+
+      whenReady(result, timeout(Span(2, Seconds))) {
+        yetuUser =>
```

Comment:
You might make this a bit simpler using the `Inside` trait and pattern matching (http://www.scalatest.org/user_guide/using_matchers#matchingAPattern):

```
inside(yetuUser) {
  case Some(YetuUser(`testUser.email`, `testUser.firstName`, `testUser.passwordInfo`, Some(userAgreement)) =>
    userAgreement.acceptTermsAndConditions mustEqual true
    userAgreement.acceptTermsAndConditionsDate.toDate mustEqual DateTime.now.toDate
}
```
When you use backquotes in a pattern match you are telling Scala to match against the value of that variable (instead of capturing the value).